### ISSUE TITLE
Weird `nil` type behaviour for some mutation in 1.5rc.

### ISSUE THREAD

**Issue by hickscorp** [absinthe-graphql/absinthe#922]
Up until now, we were using absinthe latest `pre` version and everything was working fine.
We are trying to upgrade to the RC and we're getting some cryptic error while running the following mutation:

```
mutation($input: LinkEmailInput!) {
  linkEmailAddress(input: $input) {
    linkedEmail {
      ...LinkedEmail
    }
  }
}

fragment Identifiable on Node {
  __typename
  id
}

fragment LinkedEmail on LinkedEmail {
  ...Identifiable
  address
  verifiedAt
}
```

The input looks like so:

```
{
	"input": {
		"address": "someone@somewhere.com"
	}
}
```
When running this mutation against the endpoint, we get the following stack trace:

```
Request: POST /graphiql
** (exit) an exception was raised:
    ** (UndefinedFunctionError) function nil.name/0 is undefined
        nil.name()
        (absinthe 1.5.0) lib/absinthe/phase/document/arguments/variable_types_match.ex:73: Absinthe.Phase.Document.Arguments.VariableTypesMatch.check_var_type/3
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:16: anonymous fn/3 in Absinthe.Blueprint.Transform.prewalk/2
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:109: Absinthe.Blueprint.Transform.walk/4
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:145: anonymous fn/4 in Absinthe.Blueprint.Transform.walk_children/5
        (elixir 1.10.2) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:114: Absinthe.Blueprint.Transform.walk/4
        (elixir 1.10.2) lib/enum.ex:1520: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:145: anonymous fn/4 in Absinthe.Blueprint.Transform.walk_children/5
        (elixir 1.10.2) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:114: Absinthe.Blueprint.Transform.walk/4
        (elixir 1.10.2) lib/enum.ex:1520: Enum."-map_reduce/3-lists^mapfoldl/2-0-"/3
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:145: anonymous fn/4 in Absinthe.Blueprint.Transform.walk_children/5
        (elixir 1.10.2) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:114: Absinthe.Blueprint.Transform.walk/4
        (absinthe 1.5.0) lib/absinthe/blueprint/transform.ex:15: Absinthe.Blueprint.Transform.prewalk/2
        (elixir 1.10.2) lib/enum.ex:1396: Enum."-map/2-lists^map/1-0-"/2
        (elixir 1.10.2) lib/map.ex:798: Map.update!/3
        (absinthe 1.5.0) lib/absinthe/phase/document/arguments/variable_types_match.ex:14: Absinthe.Phase.Document.Arguments.VariableTypesMatch.run/2
        (absinthe 1.5.0) lib/absinthe/pipeline.ex:368: Absinthe.Pipeline.run_phase/3
```

We tried following the stack trace down, basically in the `VariableTypesMatch` module around line 70, `var_type_schema` is equal to `%Absinthe.Type.NonNull{of_type: nil, ...}`.

Happy to share the whole schema / screenshare about this in private as the whole schema is quite complex.

This might help too:

```
  @desc "Mutations that are related to linked email objects."
  object :linked_email_mutations do
    @desc "Allows to initiate the linking of a new email address to a user."
    payload field :link_email_address do
      input do
        @desc "The email address to verify and link."
        field :address, non_null(:string)
      end

      output do
        field :linked_email, non_null(:linked_email)
      end

      resolve &AccountsResolver.link_email_address/2
    end
  end
```

**Comment by jayjun** [comment]
Encountered in 1.5.0. To reproduce, set a variable type to a non-null input object that does not exist.

```graphql
query ($episode: Episode!) {
  hero(episode: $episode) {
    name
  }
}
```
```elixir
query do
  field :hero, :hero do
    arg :episode, non_null(:episode_input)
  end
end
```

In this example, `Episode` is not defined on schema.

**Comment by hickscorp** [comment]
@jayjun Correct!

In our case, `LinkEmailInput!` is incorrect, and should read `LinkEmailAddressInput!`.

**Comment by benwilson512** [comment]
Fixed via #923, v1.5.1 is out, thanks folks!
